### PR TITLE
Kerberos "Change password" protocol

### PR DESCRIPTION
**Pull request by RRRadicalEdward**
I added some new structures, messages, and constants needed for the _Kerberos_ "_Change password_" protocol.
I also added doc comments to some other _Kerberos_ constants.

### CODE REVIEW

File: picky-krb/src/constants.rs [hunk 1]
```diff
@@ -1,4 +1,5 @@
 pub mod types {
+    /// [Kerberos Message Types](https://datatracker.ietf.org/doc/html/rfc4120#section-7.5.7)
```
Comment (CBenoit) [inline picky-krb/src/constants.rs:2]: Triple slash is used to document the next item, so either use
```suggestion
    // [Kerberos Message Types](https://datatracker.ietf.org/doc/html/rfc4120#section-7.5.7)
```
or something like
```suggestion
    //= [Kerberos Message Types](https://datatracker.ietf.org/doc/html/rfc4120#section-7.5.7) =//
```
at your preference

File: picky-krb/src/constants.rs [hunk 2]
```diff
@@ -8,8 +9,12 @@ pub mod types {
     pub const TGT_REQ_MSG_TYPE: u8 = 0x10;
     pub const TGT_REP_MSG_TYPE: u8 = 0x11;
 
+    pub const KRB_PRIV: u8 = 21;
+    pub const KRB_PRIV_ENC_PART: u8 = 28;
+
     pub const KRB_ERROR_MSG_TYPE: u8 = 0x1e;
 
+    /// [Principal Names](https://datatracker.ietf.org/doc/html/rfc4120#section-6.2)
```
Comment (CBenoit) [inline picky-krb/src/constants.rs:17]: Same as above

File: picky-krb/src/constants.rs [hunk 3]
```diff
@@ -20,6 +25,7 @@ pub mod types {
     pub const NT_SMTP_NAME: u8 = 0x07;
     pub const NT_ENTERPRISE: u8 = 0x0A;
 
+    /// [PreAuthentication Data Types](https://datatracker.ietf.org/doc/html/rfc4120#section-7.5.2)
```
Comment (CBenoit) [inline picky-krb/src/constants.rs:28]: Same as above

File: picky-krb/src/constants.rs [hunk 4]
```diff
@@ -33,24 +39,29 @@ pub mod types {
     pub const ENC_TGS_REP_PART_TYPE: u8 = 26;
     pub const ENC_AP_REP_PART_TYPE: u8 = 27;
 
+    /// [Address Types](https://datatracker.ietf.org/doc/html/rfc4120#section-7.5.3)
```
Comment (CBenoit) [inline picky-krb/src/constants.rs:42]: Same as above

File: picky-krb/src/constants.rs [hunk 5]
```diff
@@ -33,24 +39,29 @@ pub mod types {
...
+    pub const IP_V6_ADDR_TYPE: u8 = 24;
 }
 
 pub mod key_usages {
+    /// [GSS API Key Usages](https://datatracker.ietf.org/doc/html/rfc4121#section-2)
     pub const ACCEPTOR_SEAL: i32 = 22;
     pub const ACCEPTOR_SIGN: i32 = 23;
     pub const INITIATOR_SEAL: i32 = 24;
     pub const INITIATOR_SIGN: i32 = 25;
+    /// [Key Usage Numbers](https://datatracker.ietf.org/doc/html/rfc4120#section-6.2)
```
Comment (CBenoit) [inline picky-krb/src/constants.rs:60]: Same as above

File: picky-krb/src/constants.rs [hunk 6]
```diff
@@ -33,24 +39,29 @@ pub mod types {
...
+    pub const XNS_ADDR_TYPE: u8 = 6;
+    pub const ISO_ADDR_TYPE: u8 = 7;
+    pub const DECNET_PHASE_IV_ADDR_TYPE: u8 = 12;
+    pub const APPLETALK_DDP_ADDR_TYPE: u8 = 16;
+    pub const NET_BIOS_ADDR_TYPE: u8 = 20;
+    pub const IP_V6_ADDR_TYPE: u8 = 24;
 }
 
 pub mod key_usages {
+    /// [GSS API Key Usages](https://datatracker.ietf.org/doc/html/rfc4121#section-2)
```
Comment (CBenoit) [inline picky-krb/src/constants.rs:55]: Same as above

File: picky-krb/src/constants.rs [hunk 7]
```diff
@@ -33,24 +39,29 @@ pub mod types {
...
     pub const ACCEPTOR_SEAL: i32 = 22;
     pub const ACCEPTOR_SIGN: i32 = 23;
     pub const INITIATOR_SEAL: i32 = 24;
     pub const INITIATOR_SIGN: i32 = 25;
+    /// [Key Usage Numbers](https://datatracker.ietf.org/doc/html/rfc4120#section-6.2)
+    pub const KRB_PRIV_ENC_PART: i32 = 13;
```
Comment (CBenoit) [inline picky-krb/src/constants.rs:61]: I don’t see this constant in the link above (https://datatracker.ietf.org/doc/html/rfc4120#section-6.2)

**Review by CBenoit — CHANGES_REQUESTED** [review]
Thank you!

File: picky-krb/src/constants.rs (hunk 7 above)
Comment (RRRadicalEdward) [reply picky-krb/src/constants.rs:61]: Yeah. It's from a different section of the RFC. I fixed it.

Comment (RRRadicalEdward) [reply picky-krb/src/constants.rs:61]: Yeah, It's from a different section of the RFC. I fixed it.

File: picky-krb/src/constants.rs (hunk 1 above)
Comment (RRRadicalEdward) [reply picky-krb/src/constants.rs]: Fixed it from all the cases

**Review by CBenoit — APPROVED** [review]
LGTM!

**Pull request merged**